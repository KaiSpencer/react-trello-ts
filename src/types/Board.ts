import {CSSProperties} from 'react'

export interface BoardData {
  lanes: Lane[]
}

export interface Lane {
  id: string
  title?: string
  label?: string
  style?: CSSProperties
  cards: Card[]
  currentPage?: number
  droppable?: boolean
  labelStyle?: CSSProperties
  cardStyle?: CSSProperties
  disallowAddingCard?: boolean
  [key: string]: any
}

export interface Card {
  id: string
  title?: string
  label?: string
  description?: string
  laneId?: string
  style?: CSSProperties
  draggable?: boolean
  [key: string]: any
}
