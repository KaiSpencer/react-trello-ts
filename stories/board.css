.boardContainer {
    background-color: #4BBF6B;
}