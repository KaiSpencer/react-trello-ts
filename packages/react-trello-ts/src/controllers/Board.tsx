import React, { createContext, FC, PropsWithChildren } from "react";
import classNames from "classnames";
import uuidv1 from "uuid/v1";
import { BoardContainer } from "./BoardContainer";
import * as DefaultComponets from "../components";
import { BoardData } from "../types/Board";
import { store } from "../store/store";

export const Board: FC<
	PropsWithChildren<{
		id?: string;
		className?: string;
		components: typeof DefaultComponets;
		data: BoardData;
		t?: any;
	}>
> = ({ data, children, className, components, id, t, ...rest }) => {
	const allClassNames = classNames("react-trello-board", className || "");

	return (
		<BoardContext.Provider value={store}>
			<components.GlobalStyle />
			<BoardContainer
				id={id || uuidv1()}
				data={data}
				components={components}
				t={t}
				className={allClassNames}
				{...rest}
			/>
		</BoardContext.Provider>
	);
};

export const BoardContext = createContext(store);
