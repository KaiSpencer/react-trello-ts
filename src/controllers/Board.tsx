import classNames from "classnames";
import React, { FC, PropsWithChildren, createContext } from "react";
import { v1 } from "uuid";
import * as DefaultComponets from "../components";
import { store } from "../store/store";
import { BoardData } from "../types/Board";
import { BoardContainer } from "./BoardContainer";

export const Board: FC<
	PropsWithChildren<{
		id?: string;
		className?: string;
		components: typeof DefaultComponets;
		data: BoardData;
		t?: any;
	}>
> = ({ data, children, className, components, id, t, ...rest }) => {
	const allClassNames = classNames("react-trello-board", className || "");

	return (
		<BoardContext.Provider value={store}>
			{/*
			// @ts-ignore */}
			<components.GlobalStyle />
			<BoardContainer
				id={id || v1()}
				data={data}
				components={components}
				t={t}
				className={allClassNames}
				{...rest}
			/>
		</BoardContext.Provider>
	);
};

export const BoardContext = createContext(store);
