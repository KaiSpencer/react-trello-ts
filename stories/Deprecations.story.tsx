import React, {Component} from 'react'
import {storiesOf} from '@storybook/react'

import Board from '../src'

import './board.css'

const data = require('./data/base.json')

const CustomLaneHeader = props => {
  const buttonHandler = () => {
    alert(`The label passed to the lane was: ${props.label}. The lane has ${props.cards.length} cards!`)
  }
  return (
    <div>
      <header
        style={{
          borderBottom: '2px solid #c5c5c5',
          paddingBottom: 6,
          marginBottom: 10,
          display: 'flex',
          flexDirection: 'row',
          justifyContent: 'space-between'
        }}>
        <div style={{fontSize: 14, fontWeight: 'bold'}}>{props.title}</div>
        {props.label && (
          <div style={{width: '30%', textAlign: 'right', fontSize: 13}}>
            <button onClick={buttonHandler} style={{cursor: 'pointer'}}>
              ?
            </button>
          </div>
        )}
      </header>
    </div>
  )
}

class NewCard extends Component {
  updateField = (field, evt) => {
    this.setState({[field]: evt.target.value})
  }

  handleAdd = () => {
    this.props.onAdd(this.state)
  }

  render() {
    const {onCancel} = this.props
    return (
      <div style={{background: 'white', borderRadius: 3, border: '1px solid #eee', borderBottom: '1px solid #ccc'}}>
        <div style={{padding: 5, margin: 5}}>
          <div>
            <div style={{marginBottom: 5}}>
              <input type="text" onChange={evt => this.updateField('title', evt)} placeholder="Title" />
            </div>
            <div style={{marginBottom: 5}}>
              <input type="text" onChange={evt => this.updateField('description', evt)} placeholder="Description" />
            </div>
          </div>
          <button onClick={this.handleAdd}>Add</button>
          <button onClick={onCancel}>Cancel</button>
        </div>
      </div>
    )
  }
}
const CustomCard = props => {
  return (
    <div style={{padding: 6}}>
      <header
        style={{
          borderBottom: '1px solid #eee',
          paddingBottom: 6,
          marginBottom: 10,
          display: 'flex',
          flexDirection: 'row',
          justifyContent: 'space-between',
          color: props.cardColor
        }}>
        <div style={{fontSize: 14, fontWeight: 'bold'}}>{props.name}</div>
        <div style={{fontSize: 11}}>{props.dueOn}</div>
      </header>
      <div style={{fontSize: 12, color: '#BD3B36'}}>
        <div style={{color: '#4C4C4C', fontWeight: 'bold'}}>{props.subTitle}</div>
        <div style={{padding: '5px 0px'}}>
          <i>{props.body}</i>
        </div>
        <div style={{marginTop: 10, textAlign: 'center', color: props.cardColor, fontSize: 15, fontWeight: 'bold'}}>
          {props.escalationText}
        </div>
        {props.tags && (
          <div
            style={{
              borderTop: '1px solid #eee',
              paddingTop: 6,
              display: 'flex',
              justifyContent: 'flex-end',
              flexDirection: 'row',
              flexWrap: 'wrap'
            }}>
            {props.tags.map(tag => (
              <Tag key={tag.title} {...tag} tagStyle={props.tagStyle} />
            ))}
          </div>
        )}
      </div>
    </div>
  )
}
storiesOf('Deprecation warnings', module).add(
  'v2.2 warnings',
  () => (
    <Board
      data={data}
      editable
      addCardLink={<button>New Card</button>}
      customLaneHeader={<CustomLaneHeader />}
      newLaneTemplate={<div>new lane</div>}
      newCardTemplate={<NewCard />}
      customCardLayout>
      <CustomCard />
    </Board>
  ),
  {
    info:
      'Example of usage legacy props: addCardLink, customCardLayout, customLaneHeader, newLaneTemplate, newCardTemplate'
  }
)
