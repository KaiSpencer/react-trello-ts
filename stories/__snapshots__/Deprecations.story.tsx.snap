// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Deprecation warnings v2_2_warnings smoke-test 1`] = `
<div draggable="false"
     class="sc-iGgWBj iSNfA"
>
  <div>
    <div class="smooth-dnd-container horizontal">
      <section draggable="false"
               class="sc-kAyceB dgWEjS react-trello-lane"
               style="width: 280px;"
      >
        <header class="sc-gsFSXq sc-imWYAI imRIDz gLxuAH">
          <span class="sc-kpDqfm gDkBAH">
            Planned Tasks
          </span>
          <span class="sc-dAlyuH iHiWje">
            <span>
              20/70
            </span>
          </span>
        </header>
        <div class="sc-dhKdcB ikKiDc">
          <div class="smooth-dnd-container vertical">
            <div class="smooth-dnd-draggable-wrapper">
              <article data-id="Milk"
                       class="sc-jlZhew sc-cwHptR clGdKf fQNcVl react-trello-card"
              >
                <header class="sc-gsFSXq sc-jEACwC imRIDz jNaWkr">
                  <span draggable="true"
                        class="sc-kpDqfm sc-cPiKLX cFicSq cnDDQK"
                  >
                    <textarea placeholder="title"
                              autocomplete="off"
                              autocorrect="off"
                              autocapitalize="off"
                              spellcheck="false"
                              rows="1"
                              class="sc-fHjqPf bsHyvp"
                              style="overflow: hidden; overflow-wrap: break-word; height: 18px;"
                    >
                      Buy milk
                    </textarea>
                  </span>
                  <span class="sc-dAlyuH sc-dLMFU iHiWje dvkFIZ">
                    <textarea placeholder="label"
                              autocomplete="off"
                              autocorrect="off"
                              autocapitalize="off"
                              spellcheck="false"
                              rows="1"
                              class="sc-fHjqPf bsHyvp"
                              style="overflow: hidden; overflow-wrap: break-word; height: 18px;"
                    >
                      15 mins
                    </textarea>
                  </span>
                  <div class="sc-hmdomO gJuJJJ">
                    <button class="sc-jsJBEP djrAfI">
                      ✖
                    </button>
                  </div>
                </header>
                <div class="sc-eDPEul cRhiim">
                  <textarea placeholder="description"
                            autocomplete="off"
                            autocorrect="off"
                            autocapitalize="off"
                            spellcheck="false"
                            rows="1"
                            class="sc-fHjqPf bsHyvp"
                            style="overflow: hidden; overflow-wrap: break-word; height: 36px;"
                  >
                    2 Gallons of milk at the Deli store
                  </textarea>
                </div>
              </article>
            </div>
            <div class="smooth-dnd-draggable-wrapper">
              <article data-id="Plan2"
                       class="sc-jlZhew sc-cwHptR clGdKf fQNcVl react-trello-card"
              >
                <header class="sc-gsFSXq sc-jEACwC imRIDz jNaWkr">
                  <span draggable="true"
                        class="sc-kpDqfm sc-cPiKLX cFicSq cnDDQK"
                  >
                    <textarea placeholder="title"
                              autocomplete="off"
                              autocorrect="off"
                              autocapitalize="off"
                              spellcheck="false"
                              rows="1"
                              class="sc-fHjqPf bsHyvp"
                              style="overflow: hidden; overflow-wrap: break-word; height: 18px;"
                    >
                      Dispose Garbage
                    </textarea>
                  </span>
                  <span class="sc-dAlyuH sc-dLMFU iHiWje dvkFIZ">
                    <textarea placeholder="label"
                              autocomplete="off"
                              autocorrect="off"
                              autocapitalize="off"
                              spellcheck="false"
                              rows="1"
                              class="sc-fHjqPf bsHyvp"
                              style="overflow: hidden; overflow-wrap: break-word; height: 18px;"
                    >
                      10 mins
                    </textarea>
                  </span>
                  <div class="sc-hmdomO gJuJJJ">
                    <button class="sc-jsJBEP djrAfI">
                      ✖
                    </button>
                  </div>
                </header>
                <div class="sc-eDPEul cRhiim">
                  <textarea placeholder="description"
                            autocomplete="off"
                            autocorrect="off"
                            autocapitalize="off"
                            spellcheck="false"
                            rows="1"
                            class="sc-fHjqPf bsHyvp"
                            style="overflow: hidden; overflow-wrap: break-word; height: 36px;"
                  >
                    Sort out recyclable and waste as needed
                  </textarea>
                </div>
              </article>
            </div>
            <div class="smooth-dnd-draggable-wrapper">
              <article data-id="Plan3"
                       class="sc-jlZhew sc-cwHptR clGdKf fQNcVl react-trello-card"
              >
                <header class="sc-gsFSXq sc-jEACwC imRIDz jNaWkr">
                  <span draggable="true"
                        class="sc-kpDqfm sc-cPiKLX cFicSq cnDDQK"
                  >
                    <textarea placeholder="title"
                              autocomplete="off"
                              autocorrect="off"
                              autocapitalize="off"
                              spellcheck="false"
                              rows="1"
                              class="sc-fHjqPf bsHyvp"
                              style="overflow: hidden; overflow-wrap: break-word; height: 18px;"
                    >
                      Write Blog
                    </textarea>
                  </span>
                  <span class="sc-dAlyuH sc-dLMFU iHiWje dvkFIZ">
                    <textarea placeholder="label"
                              autocomplete="off"
                              autocorrect="off"
                              autocapitalize="off"
                              spellcheck="false"
                              rows="1"
                              class="sc-fHjqPf bsHyvp"
                              style="overflow: hidden; overflow-wrap: break-word; height: 18px;"
                    >
                      30 mins
                    </textarea>
                  </span>
                  <div class="sc-hmdomO gJuJJJ">
                    <button class="sc-jsJBEP djrAfI">
                      ✖
                    </button>
                  </div>
                </header>
                <div class="sc-eDPEul cRhiim">
                  <textarea placeholder="description"
                            autocomplete="off"
                            autocorrect="off"
                            autocapitalize="off"
                            spellcheck="false"
                            rows="1"
                            class="sc-fHjqPf bsHyvp"
                            style="overflow: hidden; overflow-wrap: break-word; height: 18px;"
                  >
                    Can AI make memes?
                  </textarea>
                </div>
              </article>
            </div>
            <div class="smooth-dnd-draggable-wrapper">
              <article data-id="Plan4"
                       class="sc-jlZhew sc-cwHptR clGdKf fQNcVl react-trello-card"
              >
                <header class="sc-gsFSXq sc-jEACwC imRIDz jNaWkr">
                  <span draggable="true"
                        class="sc-kpDqfm sc-cPiKLX cFicSq cnDDQK"
                  >
                    <textarea placeholder="title"
                              autocomplete="off"
                              autocorrect="off"
                              autocapitalize="off"
                              spellcheck="false"
                              rows="1"
                              class="sc-fHjqPf bsHyvp"
                              style="overflow: hidden; overflow-wrap: break-word; height: 18px;"
                    >
                      Pay Rent
                    </textarea>
                  </span>
                  <span class="sc-dAlyuH sc-dLMFU iHiWje dvkFIZ">
                    <textarea placeholder="label"
                              autocomplete="off"
                              autocorrect="off"
                              autocapitalize="off"
                              spellcheck="false"
                              rows="1"
                              class="sc-fHjqPf bsHyvp"
                              style="overflow: hidden; overflow-wrap: break-word; height: 18px;"
                    >
                      5 mins
                    </textarea>
                  </span>
                  <div class="sc-hmdomO gJuJJJ">
                    <button class="sc-jsJBEP djrAfI">
                      ✖
                    </button>
                  </div>
                </header>
                <div class="sc-eDPEul cRhiim">
                  <textarea placeholder="description"
                            autocomplete="off"
                            autocorrect="off"
                            autocapitalize="off"
                            spellcheck="false"
                            rows="1"
                            class="sc-fHjqPf bsHyvp"
                            style="overflow: hidden; overflow-wrap: break-word; height: 18px;"
                  >
                    Transfer to bank account
                  </textarea>
                </div>
              </article>
            </div>
          </div>
          <a class="sc-gFqAkR ecfzPu">
            Click to add card
          </a>
        </div>
      </section>
      <section draggable="false"
               class="sc-kAyceB dgWEjS react-trello-lane"
               style="width: 280px;"
      >
        <header class="sc-gsFSXq sc-imWYAI imRIDz gLxuAH">
          <span class="sc-kpDqfm gDkBAH">
            Work In Progress
          </span>
          <span class="sc-dAlyuH iHiWje">
            <span>
              10/20
            </span>
          </span>
        </header>
        <div class="sc-dhKdcB ikKiDc">
          <div class="smooth-dnd-container vertical">
            <div class="smooth-dnd-draggable-wrapper">
              <article data-id="Wip1"
                       class="sc-jlZhew sc-cwHptR clGdKf fQNcVl react-trello-card"
              >
                <header class="sc-gsFSXq sc-jEACwC imRIDz jNaWkr">
                  <span draggable="true"
                        class="sc-kpDqfm sc-cPiKLX cFicSq cnDDQK"
                  >
                    <textarea placeholder="title"
                              autocomplete="off"
                              autocorrect="off"
                              autocapitalize="off"
                              spellcheck="false"
                              rows="1"
                              class="sc-fHjqPf bsHyvp"
                              style="overflow: hidden; overflow-wrap: break-word; height: 18px;"
                    >
                      Clean House
                    </textarea>
                  </span>
                  <span class="sc-dAlyuH sc-dLMFU iHiWje dvkFIZ">
                    <textarea placeholder="label"
                              autocomplete="off"
                              autocorrect="off"
                              autocapitalize="off"
                              spellcheck="false"
                              rows="1"
                              class="sc-fHjqPf bsHyvp"
                              style="overflow: hidden; overflow-wrap: break-word; height: 18px;"
                    >
                      30 mins
                    </textarea>
                  </span>
                  <div class="sc-hmdomO gJuJJJ">
                    <button class="sc-jsJBEP djrAfI">
                      ✖
                    </button>
                  </div>
                </header>
                <div class="sc-eDPEul cRhiim">
                  <textarea placeholder="description"
                            autocomplete="off"
                            autocorrect="off"
                            autocapitalize="off"
                            spellcheck="false"
                            rows="1"
                            class="sc-fHjqPf bsHyvp"
                            style="overflow: hidden; overflow-wrap: break-word; height: 54px;"
                  >
                    Soap wash and polish floor. Polish windows and doors. Scrap all broken glasses
                  </textarea>
                </div>
              </article>
            </div>
          </div>
          <a class="sc-gFqAkR ecfzPu">
            Click to add card
          </a>
        </div>
      </section>
      <section draggable="false"
               class="sc-kAyceB dgWEjS react-trello-lane"
               style="width: 280px;"
      >
        <header class="sc-gsFSXq sc-imWYAI imRIDz gLxuAH">
          <span class="sc-kpDqfm gDkBAH">
            Blocked
          </span>
          <span class="sc-dAlyuH iHiWje">
            <span>
              0/0
            </span>
          </span>
        </header>
        <div class="sc-dhKdcB ikKiDc">
          <div class="smooth-dnd-container vertical">
          </div>
          <a class="sc-gFqAkR ecfzPu">
            Click to add card
          </a>
        </div>
      </section>
      <section draggable="false"
               class="sc-kAyceB dgWEjS react-trello-lane"
               style="width: 280px;"
      >
        <header class="sc-gsFSXq sc-imWYAI imRIDz gLxuAH">
          <span class="sc-kpDqfm gDkBAH">
            Completed
          </span>
          <span class="sc-dAlyuH iHiWje">
            <span>
              2/5
            </span>
          </span>
        </header>
        <div class="sc-dhKdcB ikKiDc">
          <div class="smooth-dnd-container vertical">
            <div class="smooth-dnd-draggable-wrapper">
              <article data-id="Completed1"
                       class="sc-jlZhew sc-cwHptR clGdKf fQNcVl react-trello-card"
              >
                <header class="sc-gsFSXq sc-jEACwC imRIDz jNaWkr">
                  <span draggable="true"
                        class="sc-kpDqfm sc-cPiKLX cFicSq cnDDQK"
                  >
                    <textarea placeholder="title"
                              autocomplete="off"
                              autocorrect="off"
                              autocapitalize="off"
                              spellcheck="false"
                              rows="1"
                              class="sc-fHjqPf bsHyvp"
                              style="overflow: hidden; overflow-wrap: break-word; height: 36px;"
                    >
                      Practice Meditation
                    </textarea>
                  </span>
                  <span class="sc-dAlyuH sc-dLMFU iHiWje dvkFIZ">
                    <textarea placeholder="label"
                              autocomplete="off"
                              autocorrect="off"
                              autocapitalize="off"
                              spellcheck="false"
                              rows="1"
                              class="sc-fHjqPf bsHyvp"
                              style="overflow: hidden; overflow-wrap: break-word; height: 18px;"
                    >
                      15 mins
                    </textarea>
                  </span>
                  <div class="sc-hmdomO gJuJJJ">
                    <button class="sc-jsJBEP djrAfI">
                      ✖
                    </button>
                  </div>
                </header>
                <div class="sc-eDPEul cRhiim">
                  <textarea placeholder="description"
                            autocomplete="off"
                            autocorrect="off"
                            autocapitalize="off"
                            spellcheck="false"
                            rows="1"
                            class="sc-fHjqPf bsHyvp"
                            style="overflow: hidden; overflow-wrap: break-word; height: 18px;"
                  >
                    Use Headspace app
                  </textarea>
                </div>
              </article>
            </div>
            <div class="smooth-dnd-draggable-wrapper">
              <article data-id="Completed2"
                       class="sc-jlZhew sc-cwHptR clGdKf fQNcVl react-trello-card"
              >
                <header class="sc-gsFSXq sc-jEACwC imRIDz jNaWkr">
                  <span draggable="true"
                        class="sc-kpDqfm sc-cPiKLX cFicSq cnDDQK"
                  >
                    <textarea placeholder="title"
                              autocomplete="off"
                              autocorrect="off"
                              autocapitalize="off"
                              spellcheck="false"
                              rows="1"
                              class="sc-fHjqPf bsHyvp"
                              style="overflow: hidden; overflow-wrap: break-word; height: 36px;"
                    >
                      Maintain Daily Journal
                    </textarea>
                  </span>
                  <span class="sc-dAlyuH sc-dLMFU iHiWje dvkFIZ">
                    <textarea placeholder="label"
                              autocomplete="off"
                              autocorrect="off"
                              autocapitalize="off"
                              spellcheck="false"
                              rows="1"
                              class="sc-fHjqPf bsHyvp"
                              style="overflow: hidden; overflow-wrap: break-word; height: 18px;"
                    >
                      15 mins
                    </textarea>
                  </span>
                  <div class="sc-hmdomO gJuJJJ">
                    <button class="sc-jsJBEP djrAfI">
                      ✖
                    </button>
                  </div>
                </header>
                <div class="sc-eDPEul cRhiim">
                  <textarea placeholder="description"
                            autocomplete="off"
                            autocorrect="off"
                            autocapitalize="off"
                            spellcheck="false"
                            rows="1"
                            class="sc-fHjqPf bsHyvp"
                            style="overflow: hidden; overflow-wrap: break-word; height: 18px;"
                  >
                    Use Spreadsheet for now
                  </textarea>
                </div>
              </article>
            </div>
          </div>
          <a class="sc-gFqAkR ecfzPu">
            Click to add card
          </a>
        </div>
      </section>
      <section draggable="false"
               class="sc-kAyceB dgWEjS react-trello-lane"
               style="width: 280px;"
      >
        <header class="sc-gsFSXq sc-imWYAI imRIDz gLxuAH">
          <span class="sc-kpDqfm gDkBAH">
            Repeat
          </span>
          <span class="sc-dAlyuH iHiWje">
            <span>
              1/1
            </span>
          </span>
        </header>
        <div class="sc-dhKdcB ikKiDc">
          <div class="smooth-dnd-container vertical">
            <div class="smooth-dnd-draggable-wrapper">
              <article data-id="Repeat1"
                       class="sc-jlZhew sc-cwHptR clGdKf fQNcVl react-trello-card"
              >
                <header class="sc-gsFSXq sc-jEACwC imRIDz jNaWkr">
                  <span draggable="true"
                        class="sc-kpDqfm sc-cPiKLX cFicSq cnDDQK"
                  >
                    <textarea placeholder="title"
                              autocomplete="off"
                              autocorrect="off"
                              autocapitalize="off"
                              spellcheck="false"
                              rows="1"
                              class="sc-fHjqPf bsHyvp"
                              style="overflow: hidden; overflow-wrap: break-word; height: 18px;"
                    >
                      Morning Jog
                    </textarea>
                  </span>
                  <span class="sc-dAlyuH sc-dLMFU iHiWje dvkFIZ">
                    <textarea placeholder="label"
                              autocomplete="off"
                              autocorrect="off"
                              autocapitalize="off"
                              spellcheck="false"
                              rows="1"
                              class="sc-fHjqPf bsHyvp"
                              style="overflow: hidden; overflow-wrap: break-word; height: 18px;"
                    >
                      30 mins
                    </textarea>
                  </span>
                  <div class="sc-hmdomO gJuJJJ">
                    <button class="sc-jsJBEP djrAfI">
                      ✖
                    </button>
                  </div>
                </header>
                <div class="sc-eDPEul cRhiim">
                  <textarea placeholder="description"
                            autocomplete="off"
                            autocorrect="off"
                            autocapitalize="off"
                            spellcheck="false"
                            rows="1"
                            class="sc-fHjqPf bsHyvp"
                            style="overflow: hidden; overflow-wrap: break-word; height: 18px;"
                  >
                    Track using fitbit
                  </textarea>
                </div>
              </article>
            </div>
          </div>
          <a class="sc-gFqAkR ecfzPu">
            Click to add card
          </a>
        </div>
      </section>
      <section draggable="false"
               class="sc-kAyceB dgWEjS react-trello-lane"
               style="width: 280px;"
      >
        <header class="sc-gsFSXq sc-imWYAI imRIDz gLxuAH">
          <span class="sc-kpDqfm gDkBAH">
            Archived
          </span>
          <span class="sc-dAlyuH iHiWje">
            <span>
              1/1
            </span>
          </span>
        </header>
        <div class="sc-dhKdcB ikKiDc">
          <div class="smooth-dnd-container vertical">
            <div class="smooth-dnd-draggable-wrapper">
              <article data-id="Archived1"
                       class="sc-jlZhew sc-cwHptR clGdKf fQNcVl react-trello-card"
              >
                <header class="sc-gsFSXq sc-jEACwC imRIDz jNaWkr">
                  <span draggable="true"
                        class="sc-kpDqfm sc-cPiKLX cFicSq cnDDQK"
                  >
                    <textarea placeholder="title"
                              autocomplete="off"
                              autocorrect="off"
                              autocapitalize="off"
                              spellcheck="false"
                              rows="1"
                              class="sc-fHjqPf bsHyvp"
                              style="overflow: hidden; overflow-wrap: break-word; height: 18px;"
                    >
                      Go Trekking
                    </textarea>
                  </span>
                  <span class="sc-dAlyuH sc-dLMFU iHiWje dvkFIZ">
                    <textarea placeholder="label"
                              autocomplete="off"
                              autocorrect="off"
                              autocapitalize="off"
                              spellcheck="false"
                              rows="1"
                              class="sc-fHjqPf bsHyvp"
                              style="overflow: hidden; overflow-wrap: break-word; height: 18px;"
                    >
                      300 mins
                    </textarea>
                  </span>
                  <div class="sc-hmdomO gJuJJJ">
                    <button class="sc-jsJBEP djrAfI">
                      ✖
                    </button>
                  </div>
                </header>
                <div class="sc-eDPEul cRhiim">
                  <textarea placeholder="description"
                            autocomplete="off"
                            autocorrect="off"
                            autocapitalize="off"
                            spellcheck="false"
                            rows="1"
                            class="sc-fHjqPf bsHyvp"
                            style="overflow: hidden; overflow-wrap: break-word; height: 18px;"
                  >
                    Completed 10km on cycle
                  </textarea>
                </div>
              </article>
            </div>
          </div>
          <a class="sc-gFqAkR ecfzPu">
            Click to add card
          </a>
        </div>
      </section>
      <section draggable="false"
               class="sc-kAyceB dgWEjS react-trello-lane"
               style="width: 280px;"
      >
        <header class="sc-gsFSXq sc-imWYAI imRIDz gLxuAH">
          <span class="sc-kpDqfm gDkBAH">
            Archived2
          </span>
          <span class="sc-dAlyuH iHiWje">
            <span>
              1/1
            </span>
          </span>
        </header>
        <div class="sc-dhKdcB ikKiDc">
          <div class="smooth-dnd-container vertical">
            <div class="smooth-dnd-draggable-wrapper">
              <article data-id="Archived2"
                       class="sc-jlZhew sc-cwHptR clGdKf fQNcVl react-trello-card"
              >
                <header class="sc-gsFSXq sc-jEACwC imRIDz jNaWkr">
                  <span draggable="true"
                        class="sc-kpDqfm sc-cPiKLX cFicSq cnDDQK"
                  >
                    <textarea placeholder="title"
                              autocomplete="off"
                              autocorrect="off"
                              autocapitalize="off"
                              spellcheck="false"
                              rows="1"
                              class="sc-fHjqPf bsHyvp"
                              style="overflow: hidden; overflow-wrap: break-word; height: 18px;"
                    >
                      Go Jogging
                    </textarea>
                  </span>
                  <span class="sc-dAlyuH sc-dLMFU iHiWje dvkFIZ">
                    <textarea placeholder="label"
                              autocomplete="off"
                              autocorrect="off"
                              autocapitalize="off"
                              spellcheck="false"
                              rows="1"
                              class="sc-fHjqPf bsHyvp"
                              style="overflow: hidden; overflow-wrap: break-word; height: 18px;"
                    >
                      300 mins
                    </textarea>
                  </span>
                  <div class="sc-hmdomO gJuJJJ">
                    <button class="sc-jsJBEP djrAfI">
                      ✖
                    </button>
                  </div>
                </header>
                <div class="sc-eDPEul cRhiim">
                  <textarea placeholder="description"
                            autocomplete="off"
                            autocorrect="off"
                            autocapitalize="off"
                            spellcheck="false"
                            rows="1"
                            class="sc-fHjqPf bsHyvp"
                            style="overflow: hidden; overflow-wrap: break-word; height: 18px;"
                  >
                    Completed 10km on cycle
                  </textarea>
                </div>
              </article>
            </div>
          </div>
          <a class="sc-gFqAkR ecfzPu">
            Click to add card
          </a>
        </div>
      </section>
      <section draggable="false"
               class="sc-kAyceB dgWEjS react-trello-lane"
               style="width: 280px;"
      >
        <header class="sc-gsFSXq sc-imWYAI imRIDz gLxuAH">
          <span class="sc-kpDqfm gDkBAH">
            Archived3
          </span>
          <span class="sc-dAlyuH iHiWje">
            <span>
              1/1
            </span>
          </span>
        </header>
        <div class="sc-dhKdcB ikKiDc">
          <div class="smooth-dnd-container vertical">
            <div class="smooth-dnd-draggable-wrapper">
              <article data-id="Archived3"
                       class="sc-jlZhew sc-cwHptR clGdKf fQNcVl react-trello-card"
              >
                <header class="sc-gsFSXq sc-jEACwC imRIDz jNaWkr">
                  <span draggable="true"
                        class="sc-kpDqfm sc-cPiKLX cFicSq cnDDQK"
                  >
                    <textarea placeholder="title"
                              autocomplete="off"
                              autocorrect="off"
                              autocapitalize="off"
                              spellcheck="false"
                              rows="1"
                              class="sc-fHjqPf bsHyvp"
                              style="overflow: hidden; overflow-wrap: break-word; height: 18px;"
                    >
                      Go Cycling
                    </textarea>
                  </span>
                  <span class="sc-dAlyuH sc-dLMFU iHiWje dvkFIZ">
                    <textarea placeholder="label"
                              autocomplete="off"
                              autocorrect="off"
                              autocapitalize="off"
                              spellcheck="false"
                              rows="1"
                              class="sc-fHjqPf bsHyvp"
                              style="overflow: hidden; overflow-wrap: break-word; height: 18px;"
                    >
                      300 mins
                    </textarea>
                  </span>
                  <div class="sc-hmdomO gJuJJJ">
                    <button class="sc-jsJBEP djrAfI">
                      ✖
                    </button>
                  </div>
                </header>
                <div class="sc-eDPEul cRhiim">
                  <textarea placeholder="description"
                            autocomplete="off"
                            autocorrect="off"
                            autocapitalize="off"
                            spellcheck="false"
                            rows="1"
                            class="sc-fHjqPf bsHyvp"
                            style="overflow: hidden; overflow-wrap: break-word; height: 18px;"
                  >
                    Completed 10km on cycle
                  </textarea>
                </div>
              </article>
            </div>
          </div>
          <a class="sc-gFqAkR ecfzPu">
            Click to add card
          </a>
        </div>
      </section>
    </div>
  </div>
</div>
`;
